modules/user_accounts/php/NDB_Form_user_accounts.class.inc

```diff
@@ -780,9 +784,50 @@ class NDB_Form_User_Accounts extends NDB_Form
     }
 
     /**
+     * Determines if the new password that the user entered is different
+     * than its current password (the one stored in the database).
+     *
+     * @param Database $db          database object.
+     * @param string   $newPassword the new password the user entered.
+     *
+     * @return bool  true if the password changed, false otherwise.
+     */
+    function _passwordChanged($db, $newPassword)
+    {
+        //--- Get current password stored in database
+        $passwordQuery
+            = "SELECT Password_hash, Password_md5
+               FROM users 
+               WHERE UserID = :UID";
+
+        $passwords = $db->pselectRow(
+            $passwordQuery,
+            array('UID' => $this->identifier)
+        );
+
+        // If we are using PHP5.5+ and entry Password_has in the DB is not null
+        // use method password_verify to check if the password changed
+        if (function_exists('password_verify')
+            && !is_null($passwords['Password_hash'])
+        ) {
+            return !password_verify($newPassword, $passwords['Password_hash']);
+        } elseif (!function_exists('password_verify')
+            && !is_null($passwords['Password_md5'])
+        ) {
+            // If PHP version < 5.5, use old MD5Unsalt method to check for
+            // password change
+            return !User::MD5Unsalt($newPassword, $passwords['Password_md5']);
+        }
+
+        return true;
```

Comment:
Is there ever a case where PHP 5.5+ is being used (therefore password_verify exists), but the password is still stored in Password_md5? It seems like this would only happen if passwords are changed from the backend... which I often do when I forget my password. I'm not sure if the password_hash gets reset on login or not then. Anyway, probably a pretty small case and not much to worry about.